stream_alert/athena_partition_refresh/main.py

```diff
@@ -29,25 +29,92 @@
+                 details['wait'],
+                 details['tries'],
+                 details['target'])
+
+def _success_handler(details):
+    """Backoff logging handler for when backoff succeeds.
+
+    Args:
+        details (dict): Backoff context containing the number of tries,
+            target function currently executing, kwargs, args, value,
+            and wait time.
+    """
+    LOGGER.debug('[Backoff]: Completed after %d tries calling %s',
+                 details['tries'],
+                 details['target'])
+
+def _load_config():
+    """Load the StreamAlert Athena configuration files
+
+    Returns:
+        [dict] Configuration settings by file, includes two keys:
+            lambda: All lambda function settings
+            global: StreamAlert global settings
+
+    Raises:
+        [ConfigError] For invalid or missing configuration files.
+    """
+    config_files = ('lambda', 'global')
+    config = {}
+    for config_file in config_files:
+        config_file_path = 'conf/{}.json'.format(config_file)
+
+        if not os.path.exists(config_file_path):
+            raise ConfigError('The \'{}\' config file was not found'.format(
+                config_file_path))
+
+        with open(config_file_path) as config_fh:
+            try:
+                config[config_file] = json.load(config_fh)
+            except ValueError:
+                raise ConfigError('The \'{}\' config file is not valid JSON'.format(
+                    config_file))
+
+    return config
+
+
 class ConfigError(Exception):
     """Custom StreamAlertAthena Config Exception Class"""
     pass
 
 
+class AthenaPartitionRefreshError(Exception):
+    """Generic Athena Partition Error for erroring the Lambda function"""
+    pass
+
+
 class StreamAlertAthenaClient(object):
+    """A StreamAlert Athena Client for creating tables, databases, and executing queries
+
+    Attributes:
+        config: The loaded StreamAlert configuration
+        athena_client: Boto3 Athena client
+        athena_results_bucket: The S3 bucket to store Athena query results
+        athenea_results_key: The key in S3 to store Athena query results
+    """
     DATABASE_DEFAULT = 'default'
     DATABASE_STREAMALERT = 'streamalert'
     DEFAULT_S3_PREFIX = 'athena_partition_refresh'
 
     def __init__(self, **kwargs):
         """Initialize the Boto3 Athena Client, and S3 results bucket/key
 
-        Keyword Arguments:
-            config [CLIConfig]: Loaded StreamAlert configuration (optional)
-            results_key_prefix [string]: The S3 key prefix to store Athena results (optional)
+        Keyword Args:
+            config (CLIConfig): Loaded StreamAlert configuration
+            results_key_prefix (string): The S3 key prefix to store Athena results (optional)
         """
-        # Load the config from files or accept it as an argument
-        self.config = kwargs.get('config') or self._load_config()
+        self.config = kwargs['config']
```

Comment:
I know you're a big fan of kwargs, but I worry when we use it in cases where the parameter is very much so required. For instance, there is nothing in the current interface to enforce that the initializer receives a `config=...` argument, even though this class would fail miserably without it. If someone wanted to use this class and instantiated `StreamAlertAthenaClient()` without any args, it wouldn't fail explicitly until the kwargs were accessed and wouldn't even result in a pylint error telling them something was wrong (http://pylint-messages.wikidot.com/messages:e1120).

kwargs are great for optionals and maybe this is a personal preference, but positionals help enforce required params and make the interface easier to use/understand without having to rely on an exception (when the kwargs['config'] fails) to know what's going on..

/end rant.. no need to change here, just some insight on why I tend to not use them for functions with only a handful of args.

That said - you can still call a function with the `arg=value` syntax even without using kwargs, if that's any consolation.